sambacc/varlink/keybridge.py

```
@@ -0,0 +1,498 @@
+                entry=name,
+                scope=scope,
+            )
+        _delete(name)
+
+
+def endpoint(scopes: list[KeyBridgeScope]) -> VarlinkEndpoint:
+    """Return a new endpoint for the keybridge server that will be
+    constructed using the supplied scope objects.
+    """
+    return VarlinkEndpoint(
+        interface_filename="org.samba.containers.keybridge.varlink",
+        interface_name="org.samba.containers.keybridge",
+        interface_cls=KeyBridge,
+        interface_kwargs={"scopes": scopes},
+    )
+
+
+class StaticValueScope:
+    """Keybridge scope that contains simple static values.
+    FOR TESTING ONLY.
+    """
+
+    def name(self) -> str:
+        return "static_value_scope"
+
+    def about(self) -> ScopeInfo:
+        return ScopeInfo(
+            self.name(),
+            default=False,
+            kind="test",
+            description="static value test scope",
+        )
+
+    def get(self, key: str, kind: EntryKind) -> str:
+        value = {
+            "foo": b"foo-opolis",
+            "bar": b"bar harbor",
+            "baz": b"bazlee manor",
+        }[key]
+        if kind is EntryKind.B64:
+            return base64.b64encode(value).decode()
+        elif kind in EntryKind.VALUE:
+            return value.decode()
+        raise InvalidKindError()
+
+
+class MemScope:
+    """Keybridge scope that wrapping a simple dict-based in memory store.
+    FOR TESTING ONLY.
+    """
+
+    def __init__(self) -> None:
+        self._entries: dict[str, tuple[str, EntryKind]] = {}
+        _logger.debug("Created MEM Scope: defaults=%r", self._entries)
+
+    def name(self) -> str:
+        return "mem"
+
+    def about(self) -> ScopeInfo:
+        return ScopeInfo(
+            self.name(),
+            default=False,
+            kind="test",
+            description="r/w in memory test scope",
+        )
+
+    def get(self, key: str, kind: EntryKind) -> str:
+        value, value_kind = self._entries[key]
```

shouldn't we also check for key errors here?

---

We could, especially if we want to return a custom error, but the Get method of KeyBridge does it in a generic way, so it's not necessary here.